resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -433,6 +433,16 @@ private[spark] object Config extends Logging {
       .booleanConf
       .createWithDefault(false)
 
+  val KUBERNETES_EXECUTOR_MISSING_POD_DETECT_DELTA =
+    ConfigBuilder("spark.kubernetes.executor.missingPodDetectDelta")
+      .doc("When a registered executor's POD is missing from the Kubernetes API server's polled " +
+        "list of PODs then this delta time is taken as the accepted time difference between the " +
+        "registration time and the time of the polling. After this time the POD is considered " +
+        "missing from the cluster and the executor will be removed.")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .checkValue(delay => delay > 0, s"delay must be a positive time value")
+      .createWithDefaultString("30s")
```

Comment:
Is there a reason why we choose `30s`?

Reply:
As one poll is also 30s this seemed to me a reasonable value (so we just skip one poll time with the reaction). 
So this way this is a very safe setting to avoid stopping of executors by mistake.

Reply:
Then, why don't we reuse `KUBERNETES_EXECUTOR_API_POLLING_INTERVAL` instead of defining a new constant, @attilapiros ? Technically, it sounds like we need to adjust this config together if we change `KUBERNETES_EXECUTOR_API_POLLING_INTERVAL`.
> As one poll is also 30s this seemed to me a reasonable value (so we just skip one poll time with the reaction).
So this way this is a very safe setting to avoid stopping of executors by mistake.

Reply:
I think it make sense to configure them separately here I just gave an explanation what was in my mind regarding its default.

Reply:
I think you're right, having this be configurable makes sense. In my own work I've noticed different behavior between even different versions of minikube.